File: 2_python/ch06/reverse_string_test.py
import unittest  # The test framework
from typing import List

# https://leetcode.com/problems/valid-palindrome/


def reverse_string(characters: List[str]) -> List[str]:
    characters.reverse()
    return characters


class TestReverseString(unittest.TestCase):

    def test_case1(self):
        # given
        characters = ["h", "e", "l", "l", "o"]
        # when
        res = reverse_string(characters)
        # then
        assert res == ["o", "l", "l", "e", "h"]

    def test_case2(self):
        # given
        characters = ["H", "a", "n", "n", "a", "h"]
        # when
        res = reverse_string(characters)
        # then
        assert res == ["h", "a", "n", "n", "a", "H"]


if __name__ == '__main__':
    unittest.main()

File: 2_python/ch06/reverse-string.test.js
/**
 * @param {character[]} s
 * @return {void} Do not return anything, modify s in-place instead.
 */
const reverseString = (s) => {
	return s.reverse();
};

describe(`${reverseString.name}`, () => {
	it("case1", () => {
		// given
		const characters = ["h", "e", "l", "l", "o"];
		// when
		const res = reverseString(characters);
		// then
		expect(res).toEqual(["o", "l", "l", "e", "h"]);
	});

	it("case2", () => {
		// given
		const characters = ["h", "e", "l", "l", "o"];
		// when
		const res = reverseString(characters);
		// then
		expect(res).toEqual(["o", "l", "l", "e", "h"]);
	});
});
